Kickstarter-iOS/DataSources/SettingsAccountDataSource.swift

```diff
@@ -28,10 +28,9 @@ final class SettingsAccountDataSource: ValueCellDataSource {
   func insertCurrencyCell(currency: Currency?) -> IndexPath {
     let cellValue = SettingsCurrencyCellValue(cellType: SettingsAccountCellType.currency, currency: currency )
 
-    return self.insertRow(value: cellValue,
+    return self.appendRow(value: cellValue,
```

Comment:
This should be fine given that this cell is always expected to be last unless the picker is shown and given that that is also being appended it will be after this one. The only thing that could cause an incorrect index situation would be the removal of the picker which depends on the index being 2. In this case it'll be fine and seeing as we're refactoring this soon I don't think it's a problem but I think generally I'd suggest searching for it in the array and removing it at the index that we find it at to prevent regression bugs.

Reply:
Good catch, this was actually a bug.

Reply:
Oh good then, it seemed to me like it would have worked too 😄 🤷‍♂️